Map Context

AuroTB opened the pull request:
Scripts to create the map context manually.

To run:
`roslaunch nav_main map_tagger.launch`
`rosrun map_visualizer map_tagger.py`

You can find more info on the [Navigation's Map Contextualization Wiki](https://github.com/RoBorregos/Robocup-Home/wiki/Map-Contextualizer)

Issues:
#53

Josecisneros001 commented on catkin_home/src/navigation/base_control/scripts/cmd_vel.py:
```diff
@@ -1,54 +1,52 @@
 #!/usr/bin/env python
-'''
-    Script that simulate navigation stack action of sending velocity commands.
-'''
 import rospy
 from geometry_msgs.msg import Twist
 
-def run_cmd_velocity():
-    '''
-        Main function that simultate the action of sending velocity commands.
-    '''
-    # Starts a new node.
```
Keep a description for each Python/C++ file at the top, it is important to document this, so that future people understand its purpose, the same for functions.

Josecisneros001 commented on catkin_home/src/navigation/base_control/scripts/encoders.py:
```diff
@@ -1,58 +1,57 @@
+    #Initialize Twist Message
+    enc_msgFront = StampedEncoders()
+    enc_msgBack  = StampedEncoders()
+    enc_msgFront.encoders.time_delta = 0.005
+    enc_msgBack.encoders.time_delta = 0.005
+
+    forward  = 10000
+    backward = -10000
+    stop     = 0 
```
The old values were simulating the actual encoders. Why the need of changing these values?

Josecisneros001 commented on catkin_home/src/navigation/base_control/src/base_odom.cpp:
```diff
@@ -1,208 +1,161 @@
+int wheel1_new = 0;
+int wheel2_new = 0;
+int wheel3_new = 0;
+int wheel4_new = 0;
```
The old names were better, more explanatory.

Josecisneros001 commented on catkin_home/src/navigation/base_control/src/base_odom.cpp:
```diff
@@ -1,208 +1,161 @@
-// the wheel's x-coord & the origin, and the y-coord & the origin.
-const double robot_constant = 0.25 + 0.20;
+const double encoder_resolution = 17280;
+const double k = 0.25 + 0.20; //the sum of the distance between the wheel's x-coord and the origin, and the y-coord and the origin
```
The encoder resolution of the current motors is 4320, keep that value.

Josecisneros001 commented on catkin_home/src/navigation/base_control/src/base_odom.cpp:
```diff
@@ -1,208 +1,161 @@
-        geometry_msgs::Quaternion odom_quat = tf::createQuaternionMsgFromYaw(robot_displacement.th);
+    //publish the message
+    odom_pub.publish(odom);
     
-        // Update odometry message stamp.
-        odom.header.stamp = current_time;
-        
-        // Update odometry transform stamp.
-        odom_trans.header.stamp = current_time;
-        
-        // Update odometry message position.
-        odom.pose.pose.position.x = robot_displacement.x;
-        odom.pose.pose.position.y = robot_displacement.y;
-        odom.pose.pose.orientation = odom_quat;
-        
-        // Update odometry transform position.
-        odom_trans.transform.translation.x = robot_displacement.x;
-        odom_trans.transform.translation.y = robot_displacement.y;
-        odom_trans.transform.rotation = odom_quat;
-
-        // Update odometry velocity.
-        odom.twist.twist.linear.x = robot_velocity.x;
-        odom.twist.twist.linear.y = robot_velocity.y;
-        odom.twist.twist.angular.z = robot_velocity.th;
-
-        // Publish odometry message.
-        if (publish_odom_message) {
-            odom_pub.publish(odom);
-        }
-        
-        // Publish odometry transform.
-        odom_broadcaster.sendTransform(odom_trans);
-
-        ros::spinOnce();
-        r.sleep();
-    }
+    ros::spinOnce();
+    r.sleep();
+  }
 }
```
Is this an older version of this file? Or do you really do all those changes?

Josecisneros001 commented on catkin_home/src/navigation/map_visualizer/map_display.launch:
```diff
@@ -0,0 +1,5 @@
+<launch>
+  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find config_files)/rviz/goal_tf.rviz"/>
+  <node name="map_server" pkg="map_server" type="map_server" output="screen"
+        args="$(find config_files)/occupancy_grids/TMR_2019.yaml"/>
```
Add a description for every node.

Josecisneros001 commented on catkin_home/src/navigation/map_visualizer/scripts/goal_display.py:
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+import roslib; roslib.load_manifest('visualization_marker_tutorials')
```
Add a file description and functions description to this and also the other python files.

Josecisneros001 reviewed (changes requested):
There are some comments. I am going to wait for the description of the python files and the functions to be added to check the .py files that I missed and thus check them knowing well what their purpose is.
Also i would love to see a video of this working if possible :)

AuroTB commented on catkin_home/src/navigation/base_control/launch/base_control.launch:
```diff
@@ -1,60 +0,0 @@
-        pkg="rviz" 
-        args="-d $(find base_control)/rviz/base_control.rviz" 
-    />
- </launch>
```
All these files are deprecated versions from develop, I've just updated them, there shouldn't be any problem now.

AuroTB commented:
I've updated some stripts, here's the demo video!

https://user-images.githubusercontent.com/25405578/108119443-6dbc5900-7065-11eb-92cc-7528bdd31ec5.mp4

AuroTB commented:
> I've updated some scripts, here's the demo video!
> vokoscreen-2021-02-16_14-39-54.mp4

I've updated some scripts, here's the demo video!
vokoscreen-2021-02-16_14-39-54.mp4


Delete room demo:

https://user-images.githubusercontent.com/25405578/108120397-d35d1500-7066-11eb-94fd-4c6b2057e6a3.mp4

AuroTB commented:
I had some bug trying to implement my path finding algorithm: to not delay this PR any further I added the functionality to add a path to.
When implementing on Noetic I can make a v2 with what I had in mind.

Josecisneros001 reviewed (approved):
Great Implementation! I follow up your demo and LGTM!!
